contracts/PriceOracle/PriceOracle.sol

```diff
@@ -19,76 +19,88 @@ contract PriceOracle is IPriceOracle, System, Initializable, HydraChainConnector
     mapping(address => uint256) public validatorLastVotedDay;
     mapping(uint256 => uint256) public pricePerDay;
 
+    uint256 public votingPowerPercentageNeeded;
+
     // _______________ Initializer _______________
 
     function initialize(address _hydraChainAddr, address _aprCalculatorAddr) external initializer onlySystemCall {
         __HydraChainConnector_init(_hydraChainAddr);
         __APRCalculatorConnector_init(_aprCalculatorAddr);
+        _initialize();
     }
 
-    // _______________ Modifers _______________
-
-    modifier onlyActiveValidator() {
-        if (hydraChainContract.isValidatorActive(msg.sender) == false) {
-            revert Unauthorized("INACTIVE_STAKER");
-        }
-        _;
+    function _initialize() private onlyInitializing {
+        votingPowerPercentageNeeded = 61;
     }
 
     // _______________ External functions _______________
 
     /**
      * @inheritdoc IPriceOracle
      */
-    function vote(uint256 _price) external {
-        uint256 day = getTodayIfVoteAvailable(_price);
+    function vote(uint256 price) external {
+        if (price == 0) {
+            revert InvalidPrice();
+        }
+
+        uint256 day = _getCurrentDay();
+
+        (bool canVote, string memory errMsg) = shouldVote(day);
+        if (!canVote) {
+            revert InvalidVote(errMsg);
+        }
+
         validatorLastVotedDay[msg.sender] = day;
 
-        PriceForValidator memory priceForValidator = PriceForValidator({price: _price, validator: msg.sender});
+        PriceForValidator memory priceForValidator = PriceForValidator({price: price, validator: msg.sender});
         priceVotesForDay[day].push(priceForValidator);
 
-        emit PriceVoted(_price, msg.sender, day);
+        emit PriceVoted(price, msg.sender, day);
 
-        uint256 price = _checkPriceUpdateAvailability(day);
+        uint256 availablePrice = _checkPriceUpdateAvailability(day);
```

Comment:
_calcPriceWithQuorum or something similar sounds better to me than _checkPriceUpdateAvailability